syntax = "proto3";
import "google/protobuf/empty.proto";
//python -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. ./sensor.proto
//python -m grpc_tools.protoc --mypy_out=. --proto_path=. sensor.proto   

message TempHumidity {
  float temperature = 1;
  float humidity = 2;
}
message MeteoData {
  int32 sensor_id = 1;
  TempHumidity meteo_data = 2;
  int32 timestamp = 4;
}

message PollutionData {
  float co2 = 1;
  int32 sensor_id = 2;
  int32 timestamp = 4;
}

service LoadBalancer {
  rpc ProcessMeteoData (MeteoData) returns (google.protobuf.Empty) {}
  rpc ProcessPollutionData (PollutionData) returns (google.protobuf.Empty) {}
}

